src/aggregator/aggregation/options.go

```diff
@@ -20,22 +20,51 @@
 
 package aggregation
 
-import "github.com/m3db/m3/src/metrics/aggregation"
+import (
+	"github.com/m3db/m3/src/metrics/aggregation"
+	"github.com/uber-go/tally"
+)
 
 var (
-	defaultOptions Options
+	defaultHasExpensiveAggregations = false
 )
 
 // Options is the options for aggregations.
 type Options struct {
 	// HasExpensiveAggregations means expensive (multiplication／division)
 	// aggregation types are enabled.
 	HasExpensiveAggregations bool
+	// Metrics is as set of aggregation metrics.
+	Metrics Metrics
+}
+
+// Metrics is a set of metrics that can be used by elements.
+type Metrics struct {
+	gaugeValuesOutOfOrder tally.Counter
+}
+
+// NewMetrics is a set of aggregation metrics.
+func NewMetrics(scope tally.Scope) Metrics {
+	scope = scope.SubScope("aggregation")
+	gaugeScope := scope.SubScope("gauges")
```

Comment:
nit: is this necessary?

Reply:
In the future we'll likely have non-gauge metrics here, want to namespace them.

I'll make a sub-struct to make this cleaner though.